models/amity.py

```diff
@@ -0,0 +1,655 @@
+    special_databases = [default_db_name, empty_database]
+
+    allowed_fellow_strings = ["fellow", "f"]
+    allowed_staff_strings = ["staff", "s"]
+    allowed_office_strings = ["office", "o"]
+    allowed_living_space_strings = ["living_space", "ls", "living-space"]
+    allowed_yes_strings= ["yes", "y"]
+    allowed_no_strings = ["no", "n"]
+    error_codes = {
+        1: "Room does not exist",
+        2: "Person does not exist",
+        3: "Cannot create a duplicate room",
+        4: "Cannot create a duplicate person",
+        5: "Invalid person type",
+        6: "Invalid room type",
+        7: "Invalid accommodation option",
+        8: "No room name was provided",
+        9: "Person name was not provided",
+        10: "Cannot allocate a living space to a staff member",
+        11: "Cannot add person. The room is fully occupied",
+        12: "Cannot find the file",
+        13: "The file is empty",
+        14: "Wrongly formatted file",
+        15: "Invalid characters in the filename",
+        16: "The room is empty",
+        17: "Invalid character(s) in the database name",
+        18: "Non-existent database"
+    }
+
+    def create_room(self, room_names):
+        new_rooms = []
+        if not isinstance(room_names, list):
+            return "Only a list of strings is allowed"
+        if not len(room_names):
+            return self.return_error_for_printing(self.error_codes[8])
+
+        room_names = set(room_names)
+        for room_name in room_names:
+            if not isinstance(room_name, str):
+                raise TypeError
+            room_type = "office"
+            if room_name[-3:] == "-ls":
+                room_name = room_name[:-3]
+                room_type = "living-space"
+            if room_name.lower() not in ([room.name.lower() for room in self.get_all_rooms()]):
+                try:
+                    if room_type in self.allowed_living_space_strings:
+                        new_living_space = LivingSpace(room_name)
+                        self.living_spaces.append(new_living_space)
+                        new_rooms.append(new_living_space)
+                    else:
+                        new_office = Office(room_name)
+                        self.offices.append(new_office)
+                        new_rooms.append(new_office)
+                except TypeError as e:
+                    raise(e)
+                except Exception as e:
+                    print(e)
+                    raise(e)
+            else:
+                return self.return_error_for_printing(self.error_codes[3] + " '%s'" % room_name)
+        return new_rooms
+
+    def add_person(self, first_name, last_name, type, wants_accommodation=False):
+        if not isinstance(wants_accommodation, bool):
+            return self.return_error_for_printing(self.error_codes[7] + " '%s'" % wants_accommodation)
+        try:
+            new_person = None
+            if type.lower() in self.allowed_fellow_strings:
```

Comment:
`type` is a python builtin which has been overridden. The variable name `type` is non descriptive considering these are people. consider role or designation or job_type